pkg/agent/externalnode/external_node_controller.go

```diff
@@ -0,0 +1,753 @@
+	portUUID := interfaceConfig.PortUUID
+	portName := interfaceConfig.InterfaceName
+	port, err := c.ovsBridgeClient.GetPortData(portUUID, portName)
+	if err != nil {
+		return nil, err
+	}
+
+	attachInfo := map[string]interface{}{
+		ovsExternalIDUplinkName:               port.ExternalIDs[ovsExternalIDUplinkName],
+		ovsExternalIDHostIFIndex:              port.ExternalIDs[ovsExternalIDHostIFIndex],
+		ovsExternalIDHostIFName:               port.ExternalIDs[ovsExternalIDHostIFName],
+		ovsExternalIDUplinkPort:               port.ExternalIDs[ovsExternalIDUplinkPort],
+		ovsExternalIDEntityName:               eeName,
+		ovsExternalIDEntityNamespace:          port.ExternalIDs[ovsExternalIDEntityName],
+		ovsExternalIDIP:                       strings.Join(ips, eeSplitter),
+		interfacestore.AntreaInterfaceTypeKey: interfacestore.AntreaHost,
+	}
+	err = c.ovsBridgeClient.SetPortExternalIDs(portName, attachInfo)
+	if err != nil {
+		return nil, err
+	}
+	ifIPs := make([]net.IP, 0)
+	for _, ip := range ips {
+		ifIPs = append(ifIPs, net.ParseIP(ip))
+	}
+	iface := &interfacestore.InterfaceConfig{
+		InterfaceName: interfaceConfig.InterfaceName,
+		Type:          interfacestore.ExternalEntityInterface,
+		OVSPortConfig: &interfacestore.OVSPortConfig{
+			PortUUID: interfaceConfig.PortUUID,
+			OFPort:   interfaceConfig.OFPort,
+		},
+		EntityInterfaceConfig: &interfacestore.EntityInterfaceConfig{
+			EntityName:      eeName,
+			EntityNamespace: interfaceConfig.EntityNamespace,
+			UplinkPort: &interfacestore.OVSPortConfig{
+				PortUUID: interfaceConfig.UplinkPort.PortUUID,
+				OFPort:   interfaceConfig.UplinkPort.OFPort,
+			},
+			HostIfaceIndex: interfaceConfig.EntityInterfaceConfig.HostIfaceIndex,
+		},
+		IPs: ifIPs,
+	}
+	return iface, nil
+}
+
+func (c *ExternalNodeController) removeOVSPortsAndFlows(interfaceConfig *interfacestore.InterfaceConfig) error {
+	portUUID := interfaceConfig.PortUUID
+	portName := interfaceConfig.InterfaceName
+	if err := c.ofClient.UninstallVMUplinkFlows(portName); err != nil {
+		return err
+	}
+	port, err := c.ovsBridgeClient.GetPortData(portUUID, portName)
+	if err != nil {
+		if strings.Contains(err.Error(), "not found") {
+			klog.ErrorS(err, "OVS port is not found", "port", portName)
+			return nil
+		}
+		return err
+	}
+	hostIfName := port.ExternalIDs[ovsExternalIDHostIFName]
+	uplinkIfName := port.ExternalIDs[ovsExternalIDUplinkName]
+	uplinkPortID := port.ExternalIDs[ovsExternalIDUplinkPort]
+	adapterConfig, er := util.GetUplinkConfig(hostIfName)
+	if er != nil {
+		klog.ErrorS(err, "Failed to get the configuration on the host interface", "hostInterface", hostIfName)
+		return err
+	}
+	if err := c.ovsBridgeClient.DeletePort(port.UUID); err != nil {
+		klog.ErrorS(err, "Failed to delete port on OVS", "port", hostIfName)
+		return err
+	}
+	if err := c.ovsBridgeClient.DeletePort(uplinkPortID); err != nil {
+		return err
```

Comment:
Can we add log message here similar to L655

Reply:
Sure, I have updated this part.